core/src/main/scala/shapeless/typeable.scala

```diff
@@ -396,8 +396,14 @@ class TypeableMacros(val c: blackbox.Context) extends SingletonTypeUtils {
 
         if(!pSym.isCaseClass)
           c.abort(c.enclosingPosition, s"No default Typeable for parametrized type $tpe")
-        val fields = tpe.decls.toList collect {
-          case sym: TermSymbol if sym.isVal && sym.isCaseAccessor => sym.typeSignatureIn(tpe)
+        val fields = tpe.decls.sorted collect {
```

Comment:
Because if I understand correctly the scaladoc of `MemberScopeApi.sorted`, otherwise it's not guaranteed that the members are in declaration order, and `caseClassTypeable` seem to depend on that.